File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
         auto it = max_element(candies.begin(), candies.end());
         int maxCandy = *it;

         int n = candies.size();
         vector<bool> res(n);

         for (int i = 0; i < n; i++)
           res[i] = (candies[i] + extraCandies) >= maxCandy ? true : false;

        
        return res;
    }
};

File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        
        int n = candies.length;
        int maxCandy = 0;

        for (int c: candies)
          maxCandy = Math.max(maxCandy, c);
 
        List<Boolean> res = new ArrayList<>(n);
        for (int i = 0; i < n; i++)
          res.add((candies[i] + extraCandies) >= maxCandy ? true: false);
        
        return res;
          
    }
}